Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift

```diff
@@ -106,6 +106,45 @@ public actor SwiftPMBuildSystem {
   let fileSystem: FileSystem
   private let toolchainRegistry: ToolchainRegistry
 
+  private let swiftBuildSupportsPrepareForIndexingTask = SKSupport.ThreadSafeBox<Task<Bool, Never>?>(initialValue: nil)
+
+  #if compiler(>=6.1)
+  #warning(
+    "Remove swiftBuildSupportsPrepareForIndexing when we no longer need to support SwiftPM versions that don't have support for `--experimental-prepare-for-indexing`"
+  )
+  #endif
+  /// Whether `swift build` supports the `--experimental-prepare-for-indexing` flag.
+  private var swiftBuildSupportsPrepareForIndexing: Bool {
+    get async {
+      let task = swiftBuildSupportsPrepareForIndexingTask.withLock { task in
+        if let task {
+          return task
+        }
+        let newTask = Task { () -> Bool in
+          guard let swift = await toolchainRegistry.default?.swift else {
+            return false
+          }
+
+          do {
+            let process = Process(args: swift.pathString, "build", "--help")
+            try process.launch()
+            let result = try await process.waitUntilExit()
+            guard let output = String(bytes: try result.output.get(), encoding: .utf8) else {
+              return false
+            }
+            return output.contains("--experimental-prepare-for-indexing")
```

Comment:
I actually didn't notice that it wasn't hidden from help. We should change that, in which case you will want `--help-hidden` above.